### PR TITLE
Update to use new bundling service

### PR DESCRIPTION
**Pull request by mfranz1**
This PR is the final step to integrating the new bundling service into CLARK. File downloads from the details page and the user library page now rely on LibraryService to first check for a bundle to download and then proceeds to download that bundle if it is ready. The cart component was removed because it is no longer used. In the LO builder, the navbar exit now relies on a toggle to ensure files being uploaded are done uploading before a bundling process is triggered. 

Shortcut: https://app.shortcut.com/clarkcan/story/4275/update-client-to-use-bundling-service

Add LO to user library
image

Bundle found and able to be downloaded
image

LO needs to be bundled
image

Window dialog upload confirm
image

LO Builder Exit Errors
image

### CODE REVIEW

File: src/app/cube/library/components/library-item/library-item.component.ts [hunk 1]
```diff
@@ -32,6 +43,18 @@ export class LibraryItemComponent implements OnInit {
     this.titleClicked.emit(e);
   }
 
+  toggleDownload() {
```
Comment (EVogel1999) [inline src/app/cube/library/components/library-item/library-item.component.ts:46]:
```suggestion
  /**
   * Changes the icon of the download button
   */
  toggleDownload() {
```

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 2]
```diff
@@ -88,6 +88,33 @@ export class BuilderStore {
   // false until a save operation is attempted, true after that
   public touched: boolean;
 
+  /**
+   * Public variable used in the file upload process to fire the bundling
+   * process correctly and prevent users from exiting the builder prematuraly
```
Comment (EVogel1999) [inline src/app/onion/learning-object-builder/builder-store.service.ts:93]:
```suggestion
   * process correctly and prevent users from exiting the builder prematurely
```

**Review by Cwagne17 — APPROVED** [review]
Dang that's a lot lol, good stuff

File: src/app/onion/learning-object-builder/builder-store.service.ts [hunk 3]
```diff
@@ -88,6 +88,33 @@ export class BuilderStore {
...
+   * seconds on large file uploads before the confirm dialog fires. The undefined value
+   * toggles to the 'secondClickBack' after a user attempts to leave the builder in
+   * this time frame for the window dialog to appear. 'upload' is then toggled back
+   * to 'true' so we don't perpetually keep a user on the builder. At this stage,
+   * the upload may have most likly failed based on if the user refreshes, leaves, etc...
+   * Otherwise, the file upload process will continue after the user confirms the dialog
+   * and the upload value will flip back to 'false' until the upload is complete and then
+   * 'true'.
+   */
+  public upload = '';
```
Comment (EVogel1999) [inline src/app/onion/learning-object-builder/builder-store.service.ts:116]:
```suggestion
  public upload: 'true' | 'false' | 'undefined' | 'secondClickBack' = 'true';
```

**Review by EVogel1999 — COMMENTED** [review]
Just some comments, ignore the one about the turning an observable to a promise.

**Review by EVogel1999 — APPROVED** [review]

**Review by StarChaser44 — APPROVED** [review]

**Review by jouypo23 — APPROVED** [review]

**Pull request merged**